### ISSUE TITLE
Outlook Office.context.mailbox.item.organizer returning no data/undefined on Outlook Client v2102

### ISSUE THREAD

**Issue by bconners1005** [OfficeDev/office-js#2058]
We have a client using a system in which we pull Office.context.mailbox.item.organizer to bind for display on the page.  For some reason (on both beta and release) the system will return undefined on the Outlook Windows Client, but Outlook on the Web returns the data properly.

# Provide required information needed to triage your issue

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC Desktop
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: v2102
* Operating System: v1909
* Browser (if using Office on the web): Chrome
 
## Expected behavior

When calling Office.context.mailbox.item.organizer the system should be returning the organizer information from the client on an Appointment item.

## Current behavior

The system (on the client) is returning undefined so showing no information relating to the organizer.

## Steps to reproduce

1. On a build shown above load a non-owned Appointment and call Office.context.mailbox.item.organizer
2. System will return undefined on Client but will return the proper data on Outlook on the Web

## Link to live example(s)
1. https://gist.github.com/bconners1005/0046ad490e1c6a86c45350475fd8911a

# Provide additional details

1. If run from Outlook on the Web system returns item.  If the code is run on the version above the listed version notated on the ticket the system returns as expected.  When run on the context above the return is undefined.

## Context

Unknown on specifically when this issue started on the current build.  Noticed about a week or two ago on a non-related item.

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by exextoc** [comment]
any additional information you may have on what causes this to repro may be helpful. (Does it repro on EVERY meeting, only accepted meetings, etc.) Does it repro on later builds? (if you have access to them), Does it only repro on your client's machine, or also on your machines.

We are trying to track this down locally as well, but don't have an ETA on when this may be addressed. 

Internal Tracking Id: 5350868

**Comment by bconners1005** [comment]
Hey @exextoc 
   For context:
     How often:  This happens on all meetings that are not-owned.  If a meeting is owned the system returns the data as intended (reason for this is because we don't use the Office.context.mailbox.item.organizer but instead use the getAsynch).
    Repro:  I have tried it on v2107 current updated Windows 10 build and the system works as intended.  
    Repro on other machines: It repros on any of their machines (as they are all carbon copies of one another)

**Comment by exextoc** [comment]
Sorry just to clarify, you have Outlook v2107. And the bug DOES Repro? Or it doesn't? 

Another question: 
If you right click your Calendar, then on the context menu->Properties, 

![image](https://user-images.githubusercontent.com/28549921/129814698-21aa8979-e90a-4500-a28d-74d959e2d00f.png)

Under Type: it will say (MAPI) or (REST). 

Can you let us know what this is on buggy machines, and what it is on working machines? (if possible) Or if there is no pattern.

**Comment by bconners1005** [comment]
The bug does NOT repo on the v2107 machine so it seems isolated to just the machine running v2102.  I will contact the user and have them check on the buggy machine what the return is from before.

On the machine I cannot repo it it says MAPI

**Comment by bconners1005** [comment]
@exextoc 
I am contacting them again to make sure, but the response area for them was different.  It feels like they clicked on the wrong area but the type they say does say "Folder containing Calendar Items" but no (MAPI/REST) was next to it.  Could it be old enough from another update line it would have the MAPI/REST somewhere else?

**Comment by bconners1005** [comment]
@exextoc 
  In regards to the issue and timeline.  I have a user waiting (wherein their entire organization uses the same version carbon copied to all machines). I could move forward with this being a known issue if need be since they cannot specifically update without their IT doing full clearance. 

  Has there been movement or something I can do to help with this as well?

**Comment by exextoc** [comment]
We've investigated further. This is going to require a client update to fix. There are various things that can cause this issue to repro or not repro, which means it the API may fail 100% for some users, but be intermittent for others (or always work). Upgrading to 2107 may or may not help the issue. 

Until a fix can go out, and the affected users upgrade, the workaround is to use EWS to fetch the organizer when it is not defined in the API. Here is some sample code that will return an EWS XML (that can be parsed) for the Organizer Information: 

```
var ewsId = Office.context.mailbox.item.itemId;
var request = '<soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">' +
    '  <soap:Header><t:RequestServerVersion Version="Exchange2013" /></soap:Header>' +
    '  <soap:Body>' +
    '    <m:GetItem>' +
    '      <m:ItemShape>' +
    '        <t:BaseShape>IdOnly</t:BaseShape>' +
    '        <t:AdditionalProperties>' + 
     '     <t:FieldURI FieldURI="calendar:Organizer"/>' +
     '   </t:AdditionalProperties>' +
    '      </m:ItemShape >' +
    '      <m:ItemIds>' +
    '        <t:ItemId Id="' + ewsId + '" />' +
    '      </m:ItemIds>' +
    '    </m:GetItem>' +
    '  </soap:Body>' +
    '</soap:Envelope>';

Office.context.mailbox.makeEwsRequestAsync(request, function (asyncResult) {
    if (asyncResult.status == "failed") {
        showMessage("Action failed with error: " + asyncResult.error.message);
    }
    else {
        showMessage(asyncResult.value);
    }
});
```

I do not have an update as to what build / or when a client side fix will be available, but will update when we can. 

Thank you for your help in diagnosing this issue.

**Comment by exextoc** [comment]
We have fixed this bug and the fix should be available in build 16.0.14426.10000 or later.
When an updated build is available depends on the release channel a customer is on.

**Comment by exextoc** [comment]
I regret to inform that an issue was found with the fix, and we had to pull the change. The fix is still coming, but will be in a later build.

**Comment by exextoc** [comment]
The correct build with the fix will be: 16.0.14615.10000 or later.
